libyul/Object.cpp

```diff
@@ -134,12 +134,22 @@ vector<size_t> Object::pathToSubObject(YulString _qualifiedName) const
 		auto subIndexIt = object->subIndexByName.find(YulString{currentSubObjectName});
 		yulAssert(
 			subIndexIt != object->subIndexByName.end(),
-			"Assembly object <" + _qualifiedName.str() + "> not found or does not contain code."
-		);
-		object = dynamic_cast<Object const*>(object->subObjects[subIndexIt->second].get());
-		yulAssert(object, "Assembly object <" + _qualifiedName.str() + "> not found or does not contain code.");
-		yulAssert(object->subId != numeric_limits<size_t>::max(), "");
-		path.push_back({object->subId});
+			"Assembly object <" + _qualifiedName.str() + "> not found or does not contain code.");
```

Comment:
We no longer require it to contain code so the message needs to be adjusted:
```suggestion
			"Assembly object <" + _qualifiedName.str() + "> not found.");
```